python/valis/db/queries.py

```diff
@@ -115,3 +116,50 @@ def cone_search(ra: Union[str, float], dec: Union[str, float],
         where(vizdb.SDSSidStacked.cone_search(ra, dec, radius,
                                               ra_col='ra_sdss_id',
                                               dec_col='dec_sdss_id'))
+
+
+def carton_program_list(name_type: str) -> peewee.ModelSelect:
+    """
+    Return a list of either all cartons or programs from targetdb
+
+    Parameters
```

Comment:
In addition to a list of all unique cartons and programs, I'm thinking we may also want an option to return a mapping of which cartons and programs are associated with each other.  Either a separate query function or reworked as options in here.

Here is an example function that groups peewee results by either carton or program. 
```
a = targetdb.Carton.select(targetdb.Carton.carton, targetdb.Carton.program).dicts()

def test(key):
     o={}
     kk = 'program' if key == 'carton' else 'carton'
     for k, g in itertools.groupby(sorted(a, key=lambda x: x[key]), key=lambda x: x[key]):
         o[k] = set(i[kk] for i in g)
     return o
```

Reply:
what do you think of a new route under the `/query/list` that returns this mapping?  I think it'd be useful to be able to retrieve a maps of unique programs -> cartons, and vice versa.  I'm imagining an endpoint that takes as input either `carton` or `program` and returns the relevant mapping, e.g. for `program`
```
{...,
 'mwm_wd': {'mwm_wd_core',
  'mwm_wd_gaia',
  'mwm_wd_gaia_boss',
  'mwm_wd_pwd',
  'mwm_wd_pwd_boss'},
  ...,}
```
Maybe this direction is all we need?  Can I carton belong to more than one program?  Maybe it could be called `/query/list/program-map`?

Reply:
Just for completion, I set up the query function to go either way, but for the endpoint I just had it return the mapping of cartons in a list of programs. This is implemented here: https://github.com/sdss/valis/pull/6/commits/a17415a7d8c9dd887a274a97840c2ddaa11115ba, let me know what you think.

Reply:
I think that makes sense and this all looks good to me!